### ISSUE TITLE
Origin in split view on iPad

### ISSUE THREAD

**Issue by firvorski** [daisuke310vvv/PopupController#48]
Hi, 

I am trying to run it on an iPad split view, like Settings. However, when the origin is calculated, it seems that the width of the iPad screen is used `size: CGSize = UIScreen.main.bounds.size`, rather than the width of the view, which would be much narrower. This causes the popup to go off screen. 

I have found this enum (below) which seems responsible for the origin of the popup but I can't see how I could configure the behaviour to work with the split view. Do you maybe have any pointers?

Thanks!

`
public enum PopupLayout {
        case top, center, bottom

        func origin(_ view: UIView, size: CGSize = UIScreen.main.bounds.size) -> CGPoint {
            switch self {
            case .top: return CGPoint(x: (size.width - view.frame.width) / 2, y: 0)
            case .center: return CGPoint(x: (size.width - view.frame.width) / 2, y: (size.height - view.frame.height) / 2)
            case .bottom: return CGPoint(x: (size.width - view.frame.width) / 2, y: size.height - view.frame.height)
            }
        }
    }
`

**Comment by mureatencio** [comment]
Had the same problem, so I added an additional case called "left", that aligns the view next to the master screen.
`  public enum PopupLayout {
        case top, center, bottom, left
        
        func origin(_ view: UIView, size: CGSize = UIScreen.main.bounds.size) -> CGPoint {
            switch self {
            case .top: return CGPoint(x: (size.width - view.frame.width) / 2, y: 0)
            case .center: return CGPoint(x: (size.width - view.frame.width) / 2, y: (size.height - view.frame.height) / 2)
            case .bottom: return CGPoint(x: (size.width - view.frame.width) / 2, y: size.height - view.frame.height)
            case .left: return CGPoint(x: 0, y: 0)
            }
        }
    }`
then I customized my popup controller with that layout 
`.layout(.left)`

**Comment by firvorski** [comment]
Yes, I see. And then you just set the width to the view width and it will
fit. Thanks very much for sharing that!
